test/test_pydantic_redis.py

```diff
@@ -121,6 +120,54 @@ def test_insert_single_nested(store):
     assert books[0].author == author_as_model
 
 
+@pytest.mark.parametrize("store", redis_store_fixture)
+def test_update_nested_list_of_models(store):
+    Library.insert(library)
+    books = Book.select()
+    assert type(books) is list and len(books) == 4
+
+    _library = Library.select()[0]
+    assert all(isinstance(x, Book) for x in _library.books)
+
+
+@pytest.mark.parametrize("store", redis_store_fixture)
+def test_update_nested_set_of_models(store):
+    BooksToRead.insert(books_to_read)
+    books: List[Book] = Book.select()
+    assert type(books) is list and len(books) == 3
+    assert all(book.author.name == "Charles Dickens" for book in books)
+
+    _books_to_read = BooksToRead.select()[0]
+    assert isinstance(_books_to_read.books, set)
+    assert len(_books_to_read.books) == 3
+    assert all(isinstance(x, Book) for x in _books_to_read.books)
+
+
+@pytest.mark.parametrize("store", redis_store_fixture)
+def test_update_nested_tuple_of_models(store):
+    Basket.insert(basket)
+    books: List[Book] = Book.select()
+    assert type(books) is list and len(books) == 3
+    _basket = Basket.select()[0]
+    assert isinstance(_basket.books, tuple)
+    assert len(_basket.books) == 3
+    assert all(isinstance(x, Book) for x in _basket.books)
+
+@pytest.mark.parametrize("store", redis_store_fixture)
+def test_update_nested_tuple_of_different_models(store):
```

Comment:
Am I going too far at supporting Tuples containing different models ? :) It is a little bit more chalenging in the code as you would have to select items one by one. That would kill performances for large tuples

Reply:
Tuples can be dealt with in another pull request. Let's focus on `List` for now.